src/ir/comp.rs

```diff
@@ -446,8 +451,8 @@ impl CompInfo {
     }
 
     /// Get this type's destructor.
-    pub fn destructor(&self) -> &Option<ItemId> {
-        &self.destructor
+    pub fn destructor(&self) -> Option<(bool, ItemId)> {
```

Comment:
That `bool` is a little gross. As a calller of this API, now I have to remember wait does `true` mean virtual or not? Why not reuse `MethodKind` here?

Also, the doc comment should be updated to describe the extra information exposed.